### PR TITLE
Expanded view setup

### PR DESCRIPTION
**Pull request by alizehrehman**
- Added 2.5 zoom and pan functionality
- Added up/down scroll functionality on Thumbnail List (still has a bug and might need to be refactored later)
- Added right/left picture scroll functionality
- Updated CSS to enable functionality

### CODE REVIEW

**Comment by alizehrehman** [comment]
@tchitrakorn Can you review my PR? Thank you!

File: client/product_details/ExpandedView.jsx [hunk 1]
```diff
@@ -1,24 +1,94 @@
...
+      zoomView: false
+    };
+    
+    this.toggleZoomView = this.toggleZoomView.bind(this);
+    this.handleMouseMove = this.handleMouseMove.bind(this);
+  }
+  
+  componentDidMount() {
+    var element = document.getElementById('expandedView');
+    console.log(element);
```
Comment (tchitrakorn) [inline client/product_details/ExpandedView.jsx]: May want to remove this eventually!

File: client/product_details/ExpandedView.jsx [hunk 2]
```diff
@@ -1,24 +1,94 @@
...
+    element.addEventListener("mousemove", this.handleMouseMove);
+  }
+
+  componentWillUnmount() {
```
Comment (tchitrakorn) [inline client/product_details/ExpandedView.jsx:20]: I rarely use componentWillUnmount. Took me a while to understand when this is being executed!

File: client/product_details/ExpandedView.jsx [hunk 3]
```diff
@@ -1,24 +1,94 @@
...
+    var element = document.getElementById('expandedView');
+    console.log(element);
```
Comment (tchitrakorn) [inline client/product_details/ExpandedView.jsx]: May want to remove this eventually!

File: client/product_details/ExpandedView.jsx [hunk 4]
```diff
@@ -1,24 +1,94 @@
...
+    console.log('e offset X: ', -e.offsetX);
+    console.log('e offset Y: ', -e.offsetY);
+    console.log('x%: ', (e.offsetX / element.clientWidth) * 100);
+    console.log('y%: ', (e.offsetY / element.clientHeight) * 100);
+    console.log('background size: ', element.style.backgroundSize);
+    var xPercent = (e.offsetX / element.clientWidth) * 100;
+    var yPercent = (e.offsetY / element.clientHeight) * 100;
+    element.style.backgroundPositionX = xPercent + "%";
+    element.style.backgroundPositionY = yPercent + "%";
+  }
```
Comment (tchitrakorn) [inline client/product_details/ExpandedView.jsx]: Aside from removing some of the console.log that you might no longer need, I just want to say that this looks awesome! I didn't know about all these properties before (e.g., offset and clientWidth/Height) Very neat how they're used together for this feature. I tried to do a little bit of reading on dimensions of elements to understand why you chose to use certain properties over the others, but I'm not going to lie that these little nuances still escaped me. Would be great if you could share some of the resources you used to understand this concept!

File: client/product_details/ExpandedView.jsx [hunk 5]
```diff
@@ -1,24 +1,94 @@
...
+  render() {
+    if (this.props.selectedStyle && !this.state.zoomView) {
+      var photoGallery = this.props.selectedStyle.photos;
+      return (
+        <div id="expandedView">
+          <a target="_blank">
+            <img
+              onClick={this.toggleZoomView}
+              src={photoGallery[this.props.currPhotoIndex].url} 
+              alt={this.props.selectedStyle.name}
```
Comment (tchitrakorn) [inline client/product_details/ExpandedView.jsx:48]: Just want to add that it's great that you take care of the "alt" for all the images! A great reminder for myself that I need to do it as well.

File: client/product_details/Thumbnail.jsx [hunk 6]
```diff
@@ -7,7 +7,7 @@ const Thumbnail = (props) => {
         <img 
           src={props.photo.thumbnail_url || 'https://www.thermaxglobal.com/wp-content/uploads/2020/05/image-not-found.jpg'} 
           alt={props.name} 
-          className="stylePhoto selectedThumbnail" 
+          className={'stylePhoto' + props.view + ' selectedThumbnail'}
```
Comment (tchitrakorn) [inline client/product_details/Thumbnail.jsx:10]: I think it's really neat how you customize the css class like this :D

File: client/product_details/ThumbnailList.jsx [hunk 7]
```diff
@@ -1,33 +1,80 @@
...
+        <div id="thumbnailList" className={this.props.view}>
+          {this.props.selectedStyle.photos.length > 7 ? 
+            <a onClick={this.handleUpScroll} id="upScroll-container">
+              <i id="upScroll" class="fa fa-angle-up"></i>
+            </a> : null}    
+          <br></br>
+          {photos.map((photo, index) => {
+            var initIndex = index;
+            console.log('initialIndex: ', initIndex);
+            console.log('length: ', this.props.selectedStyle.photos.length);
```
Comment (tchitrakorn) [inline client/product_details/ThumbnailList.jsx]: May want to remove these eventually!

File: client/product_details/ThumbnailList.jsx [hunk 8]
```diff
@@ -1,33 +1,80 @@
...
+            var finalIndex;
+            if ((initIndex + this.state.firstPhotoIndex) > this.props.selectedStyle.photos.length) {
+              finalIndex = initIndex + this.state.firstPhotoIndex - this.props.selectedStyle.photos.length - 2;
+            } else {
+              finalIndex = initIndex + this.state.firstPhotoIndex;
+            }
+            console.log('finalIndex: ', finalIndex);
```
Comment (tchitrakorn) [inline client/product_details/ThumbnailList.jsx]: May want to remove this eventually!

**Review by tchitrakorn — APPROVED** [review]
Wow, it did took me quite a while to go through and understand what you implemented, but I have to say that many of the pieces are really neat and awesome! Going through your image zooming and changing logic was a great learning experience for me. 

There are several console.log that might need to be removed eventually. Another minor thing that I would like to point out is the overall structure of your product_detail folder. I think this is mainly a matter of preference, but considering that you have a lot of sub-components, would it be easier to have ProductDetails at the top level and then the rest in a /components sub-folder? At this point, I don't think it would make a lot of sense to change anything but it's just one thing that came to my mind when I was trying to understand where the props are flowing in from. Otherwise, everything looks fantastic!

**Comment by alizehrehman** [comment]
@tchitrakorn Thank you for the review! I just got rid of all the console.log statements. Thank you so much for letting me know where these were, I tried to do cmd+F and it wasn't showing me them all for some reason. Pushing up an updated commit and then I will merge :)

**Pull request merged**